### PR TITLE
[Refactoring] Replace `Paths.get` with `Path.of`

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
It is recommended to use `Path.of` instead of `Paths.get`. 

> If we’re working with a version between Java 7 and 10, we have no other choice but to use Paths.get(). Otherwise, if we’re working with a later version, we should go for Path.of(). The Paths class may indeed be deprecated in a future Java release, as it’s stated in the class’ comments. Moreover, using directly the factory method from Path spares an additional input.

https://www.baeldung.com/java-paths-get-path-of#which-one-should-we-use

Other Improvements:
- `Path.of("a", "b", "c")` can be replaced with `Path.of("a/b/c")`. The Path API is handling OS specific handling.
- `path.resolve("a").resolve("b").resolve("c")` can be replaced with `path.resolve("a/b/c")`.

Both improvements are shorter and can be copy-pasted as Linux paths.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/AbstractTypesTest.java [hunk 1]
```diff
@@ -20,24 +20,25 @@
 import io.ballerina.compiler.internal.parser.ParserRuleContext;
 import io.ballerinalang.compiler.parser.test.ParserTestUtils;
 
-import java.nio.file.Paths;
+import java.nio.file.Path;
+
```
Comment (tharindu-nw) [inline compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/AbstractTypesTest.java:24]: Remove new line

File: compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/tree/DeprecatedSyntaxAPITest.java [hunk 2]
```diff
@@ -23,7 +23,8 @@
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
-import java.nio.file.Paths;
+import java.nio.file.Path;
+
```
Comment (tharindu-nw) [inline compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/tree/DeprecatedSyntaxAPITest.java:27]: Remove new line

File: misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java [hunk 3]
```diff
@@ -226,7 +225,7 @@ private static void writeAPIDocs(ModuleLibrary moduleLib, Path output, boolean i
     }
 
     private static void copyDocerinaUI(Path output) {
-        File source = Path.of(System.getProperty("ballerina.home"), "lib", "tools", "doc-ui").toFile();
+        File source = Path.of(System.getProperty("ballerina.home"), "lib/tools/doc-ui").toFile();
```
Comment (tharindu-nw) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java:228]: Shall we keep the string sequence as parameters for consistency?

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/CheckingActionsTest.java [hunk 4]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "checking").toString();
+        return Path.of("actions/checking").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/CheckingActionsTest.java:45]: Shall we keep the string sequence as parameters for consistency?

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/FlushActionsTest.java [hunk 5]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "flush").toString();
+        return Path.of("actions/flush").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/FlushActionsTest.java:45]: Leave the parameters as is for consistency

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/QueryActionsTest.java [hunk 6]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "query").toString();
+        return Path.of("actions/query").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/QueryActionsTest.java:45]: Leave the parameters as is for consistency

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/SendReceiveActionsTest.java [hunk 7]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "send-receive").toString();
+        return Path.of("actions/send-receive").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/SendReceiveActionsTest.java:45]: Leave the parameters as is for consistency

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/StartActionsTest.java [hunk 8]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "start").toString();
+        return Path.of("actions/start").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/StartActionsTest.java:45]: Leave the parameters as is for consistency

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/TrapActionsTest.java [hunk 9]
```diff
@@ -42,6 +42,6 @@ public Object[][] dataProvider() {
 
     @Override
     public String getTestResourceDir() {
-        return Paths.get("actions", "trap").toString();
+        return Path.of("actions/trap").toString();
```
Comment (tharindu-nw) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/actions/TrapActionsTest.java:45]: Same as above. Please check the other similar instances as well

**Comment by Shadow-Devil** [comment]
Fixed review suggestions in [52a7874](https://github.com/ballerina-platform/ballerina-lang/pull/43093/commits/52a7874c416b7048e95a50a6e51e64a48d14b407)

**Comment by Shadow-Devil** [comment]
@gimantha Thank you for fixing the merge conflicts 😄

**Review by gimantha — APPROVED** [review]

**Pull request merged**